test/client/fixtures/sandbox/event/focus-blur-change-test.js

```diff
@@ -261,6 +261,22 @@ asyncTest('without handlers', function () {
     testFocusing(0, startNext);
 });
 
+asyncTest('onfocus() must not be called on disabled input', function () {
+    var focused = false;
+    var onfocus = function () {
```

Comment:
Also there is a non bubbling equivalent of `focus` event - `focusin`.
Check a native browser behavior:   Does disable input raise a `focusin` event?

Reply:
Yes, these changes also avoid "onfocusin" handler calling. Example:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<form id="parent-form">
    <input id="child-input" value="input">
</form>
<script>
    var parentForm      = document.getElementById('parent-form');
    var childInput      = document.getElementById('child-input');
    childInput.disabled = true;

    parentForm.addEventListener('focusin', function () {
        console.log('focusin');
    });

    childInput.focus();
</script>
</body>
</html>
```